plan/expression_rewriter.go

```diff
@@ -122,110 +128,183 @@ func (er *expressionRewriter) Enter(inNode ast.Node) (retNode ast.Node, skipChil
 	return inNode, false
 }
 
+func boolToExprConstant(b bool, tp *types.FieldType) *expression.Constant {
+	v := 0
+	if b {
+		v = 1
+	}
+	return &expression.Constant{Value: types.NewIntDatum(int64(v)), RetType: tp}
```

Comment:
How about tp.Tp is TypeFloat? Value won't match RetType. Will where be any problem?